### ISSUE TITLE
Please describe optional TRIM operations needed if someone prefers it over leaking SSD block structure used

### ISSUE THREAD

**Issue by t00** [Ataraxxia/secure-arch#4]
Having TRIM on any SSD gives a great performance gain when adding new data and more importantly prolongs the life of flash memory.
Drawback is that disk structure is exposed but if attacker knows the typical filesystem block locations it does not make much difference when they are determined to brute force encryption knowing the filesystem block structure - just a few guesses - ext4 header, LVM header and some fingerprints at the end.

It would be great, if you just can add discard commands as alternatives in a script.
Example articles on the subject:

https://www.guyrutenberg.com/2021/11/23/enable-trim-on-external-luks-encrypted-drive/
https://confluence.jaytaala.com/display/TKB/Enable+periodic+TRIM+-+including+on+a+LUKS+partition

**Comment by Ataraxxia** [comment]
Sorry for the long wait, but i'll check it out soon and update the tutorial

**Comment by Ataraxxia** [comment]
2 years later and I've finally sat down to this.
tl;dr i've added instruction to enable fstrim.timer

as for ciphers, even 4 years ago aes-cbc-essiv was the default:
https://security.stackexchange.com/questions/64151/xts-vs-aes-cbc-with-essiv-for-file-based-filesystem-encryption

And those considerations only have slightest sense if your laptop/pc is stolen from you periodically and attacker can compare whole drive images over a period of time without you noticing.
